src/target/adiv5.c

```diff
@@ -351,8 +351,10 @@ static uint32_t cortexm_initial_halt(ADIv5_AP_t *ap)
 			} else {
 				trncnt = 0xfff;
 			}
-			dhcsr = adiv5_dp_low_access(
+			adiv5_dp_low_access(
```

Comment:
It is no a bug, it is a feature. When running the while loop for the second time, we get the real data. If the device is halted in the first loop, your approach with 4 transactions is better. If the device is hard to catch, like a STM32F7 mostly in WFI, the original loop with 3 transactiond gives less dead time where we do not hammer on DHCSR. The shrt loop is good tested annd allows me to hot plug to a STM32F7 mostly in sleep where most other debugger do not catch the device and require cold plug(connect under reset)

Reply:
This makes the STM32 detection problems we've been having even worse because of the rapid turn-arounds. Not checking the result of advi5_dp_low_access the first time is a bug and additional problems waiting to happen.

Reply:
Where did you see problems? When? How to reproduce?
Rapid turnaround is no problem here. It is needed to have a high probability to catch the target in a small moment of awareness. We repeat the loop as long as the result makes no sense. We check that it is not all ones or zeros. This code exposes other problem,  see #1071, but that is no fault of this code.

Reply:
In response to a user on Discord who cannot see any of their STM32 targets after going from v1.7.1 -> v1.8 and with our own Nucleo hardware.

Reply:
I have tested with BMP native, StlinkV2/3, stlinkv2/3 firmware and f072. No problem seen here. Please show "blackmagic -t" on an affected board!

Reply:
Now there's part of the rub.. BMDA does not reproduce the issues because its turnarounds are slowed down sufficiently by the USB stack of the host. It only repros on real hardware and at that only on certain choices and combinations. This isn't the place to talk about this as it's muddying the contributor's PR and it's something Esden and us are already tracking and working on.